experiment/manage_release_versions.py

```diff
@@ -0,0 +1,111 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This script rotates the version strings in various below listed config files.
+Run this script when you want to move the jobs to the new support cadence.
+
+Usage:
+    1. update the Version Class below with the appropriate versions.
+    2. bazel run //experiment:manage_release_versions
+    3. for some config files you may need to make more manual changes. These
+    are described as comments before config files are defined below.
+
+"""
+
+import re
+import collections
+
+# Probably should just read out of a config file in the future.
+
+
+class Version(object):
+    # CURRENT is also termed as beta.
+    CURRENT = "1.13" # also known as beta in some other places.
+    STABLE1 = "1.12"
+    STABLE2 = "1.11"
+    STABLE3 = "1.10"
+    DEPRECATED = "1.9"
+
+# ordered-dict that keeps the mapping of the rotations.
+# we need a ordered dict to make sure we rotate in the
+# order of latest to oldest.
+RELEASE_VERSION_ROTATION_MAP = collections.OrderedDict([
+    (Version.STABLE1, Version.CURRENT),
+    (Version.STABLE2, Version.STABLE1),
+    (Version.STABLE3, Version.STABLE2),
+    (Version.DEPRECATED, Version.STABLE3),
+])
+# List of job config locations that need version rotation.
+# We might have to do more than just rotation in these job configs.
+# Assuming that the script is executed from the test-infra root
+# directory when run directly.
+#
+
+JOB_LIST = [
+    "README.md",
+    "config/jobs/kubernetes/sig-aws/kops/kops-presubmits.yaml",
+    #"config/jobs/kubernetes-security/generated-security-jobs.yaml",
+    #"config/jobs/kubernetes/generated/generated.yaml",
+    "config/jobs/kubernetes/sig-gcp/gce-conformance.yaml",
+    "config/jobs/kubernetes/sig-gcp/gpu/sig-gcp-gpu-presubmit.yaml",
+    "config/jobs/kubernetes/sig-gcp/sig-gcp-gce-config.yaml",
+    "config/jobs/kubernetes/sig-node/node-kubelet.yaml",
+    "config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml",
+    "config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-upgrade.yaml",
+    "config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-x-on-y.yaml",
+    "config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm.yaml",
+    "config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml",
+    "config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml",
+    "config/jobs/kubernetes/sig-testing/bazel-build-test.yaml",
+    "config/jobs/kubernetes/sig-testing/typecheck.yaml",
+    "config/jobs/kubernetes/sig-testing/verify.yaml",
```

Comment:
consider everything under config/jobs/kubernetes?

Reply:
we do not want to do all the jobs under there. Only a few that we care for. We might have to use skiplist otherwise which could be of the same size. 

I am actually of the opinion to move all the constants here out to another config file. That is for a future change. Do you have any comments?